### PR TITLE
Add CC3235S and CC3235S LaunchXL support

### PR DESCRIPTION
**Pull request by ghost**
This adds support for the CC3235S SOC and the corresponding Board. It is in many
aspects similar to the already supported CC3235SF. The main difference is, that
it does not have a XIP-Flash. That means the ROM-Bootloader, which loads the
data from the external serial flash to the SRAM, needs the first 16 KiB of SRAM
when bootstrapping. This has of course implications on the RAM sections, ie.
the first 16 KiB cannot be loaded and also on debugging,
s. `CONFIG_CC3235S_DEBUG`.

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
Looks like you included too many commits in this PR. Please rebase on main.

**Comment by ithinx-io** [comment]
Sorry, I was on a release tag.

File: boards/arm/cc3235s_launchxl/pinmux.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#define I2C_CC32XX_PIN_01_I2C_SCL  0x100  /*!< PIN 1 is used for I2C_SCL */
+#define I2C_CC32XX_PIN_02_I2C_SDA  0x101  /*!< PIN 2 is used for I2C_SDA */
+#define I2C_CC32XX_PIN_03_I2C_SCL  0x502  /*!< PIN 3 is used for I2C_SCL */
+#define I2C_CC32XX_PIN_04_I2C_SDA  0x503  /*!< PIN 4 is used for I2C_SDA */
+#define I2C_CC32XX_PIN_05_I2C_SCL  0x504  /*!< PIN 5 is used for I2C_SCL */
+#define I2C_CC32XX_PIN_06_I2C_SDA  0x505  /*!< PIN 6 is used for I2C_SDA */
+#define I2C_CC32XX_PIN_16_I2C_SCL  0x90F  /*!< PIN 16 is used for I2C_SCL */
+#define I2C_CC32XX_PIN_17_I2C_SDA  0x910  /*!< PIN 17 is used for I2C_SDA */
+
+int pinmux_initialize(const struct device *port)
```
Comment (gmarull) [inline boards/arm/cc3235s_launchxl/pinmux.c:69]: gentle reminder: this should be handled by a pinctrl driver https://github.com/zephyrproject-rtos/zephyr/issues/39740

**Comment by ithinx-io** [comment]
The platform is simplelink, not NXP.

**Comment by ithinx-io** [comment]
I ran twister for on device testing, but I got a built error for stackprot:
```
Including boilerplate (Zephyr base): /home/brmo/projects/zephyr-project/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/brmo/projects/zephyr-project/zephyr/tests/kernel/mem_protect/stackprot
-- Zephyr version: 2.7.99 (/home/brmo/projects/zephyr-project/zephyr), build: v2.7.99-1656-gf862f246cca0
-- Found Python3: /usr/bin/python3.9 (found suitable exact version "3.9.2") found components: Interpreter
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
-- Board: cc3235s_launchxl
-- Cache files will be written to: /home/brmo/.cache//zephyr
-- Using toolchain: zephyr 0.13.1 (/home/brmo/.local/opt/zephyr-sdk-0.13.1)
-- Found dtc: /home/brmo/.local/opt/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/brmo/projects/zephyr-project/zephyr/boards/arm/cc3235s_launchxl/cc3235s_launchxl.dts
-- Generated zephyr.dts: /home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/dts.cmake
Parsing /home/brmo/projects/zephyr-project/zephyr/Kconfig
Loaded configuration '/home/brmo/projects/zephyr-project/zephyr/boards/arm/cc3235s_launchxl/cc3235s_launchxl_defconfig'
Merged configuration '/home/brmo/projects/zephyr-project/zephyr/tests/kernel/mem_protect/stackprot/prj.conf'
Configuration saved to '/home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/.config'
Kconfig header saved to '/home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot/zephyr/include/generated/autoconf.h'
/home/brmo/.local/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gdb: error while loading shared libraries: libpython3.8.so.1.0: cannot open shared object file: No such file or directory
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/brmo/.local/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at /home/brmo/projects/zephyr-project/zephyr/subsys/random/CMakeLists.txt:11 (message):


      Warning: CONFIG_TEST_RANDOM_GENERATOR is not a truly random generator.
      This capability is not secure and it is provided for testing purposes only.
      Use it carefully.


-- Configuring done
-- Generating done
-- Build files have been written to: /home/brmo/projects/zephyr-project/zephyr/twister-out/cc3235s_launchxl/tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot
[  0%] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[  2%] Built target parse_syscalls_target
[  3%] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[  3%] Built target kobj_types_h_target
[  4%] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[  4%] Built target syscall_list_h_target
[  5%] Generating include/generated/driver-validation.h
[  5%] Built target driver_validation_h_target
[  6%] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
arm-zephyr-eabi-gcc: error: unrecognized command-line option '-mstack-protector-guard=global'; did you mean '-fstack-protector-all'?
gmake[2]: *** [zephyr/CMakeFiles/offsets.dir/build.make:76: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2612: zephyr/CMakeFiles/offsets.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
```

File: boards/arm/cc3235s_launchxl/pinmux.c (hunk 1 above)
Comment (ithinx-io) [reply boards/arm/cc3235s_launchxl/pinmux.c:69]: Ok,will do

**Comment by ithinx-io** [comment]
Is it maybe better to start over again? I think I really messed up the CI here.

**Comment by carlescufi** [comment]
> Is it maybe better to start over again? I think I really messed up the CI here.

No, just rebase on top of the current `main` branch, no need to close the PR and restart.

**Comment by cfriedt** [comment]
Nice! One of these days, I hope TI puts additional effort into getting SimpleLink BLE supported in the Zephyr split-stack LL.

File: boards/arm/cc3235s_launchxl/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# TI SimpleLink CC3235S LaunchXL Board
+
+# Copyright (c) 2019, Texas Instruments Incorporated
```
Comment (cfriedt) [inline boards/arm/cc3235s_launchxl/Kconfig.board:3]: @ithinx-io - are you an employee of TI? If not, I would suggest adding your own name to any copyrights.

**Comment by cfriedt** [comment]
I think the failing tests are likely unrelated. @ithinx-io - you may need to rebase and force-push to retrigger CI.

File: boards/arm/cc3235s_launchxl/Kconfig.board (hunk 2 above)
Comment (mbrunnen) [reply boards/arm/cc3235s_launchxl/Kconfig.board:3]: Ok, no, I am not. I will fix it.

**Comment by mbrunnen** [comment]
Thanks, right now just pinctrl is a bit blocking me.

**Comment by cfriedt** [comment]
@ithinx-io - any thoughts on adding pinctrl support? @vaishnavachath - are you able to offer any advice there?

**Comment by vaishnavachath** [comment]
> @ithinx-io - any thoughts on adding pinctrl support? @vaishnavachath - are you able to offer any advice there?

@ithinx-io You can take reference from https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_cc13xx_cc26xx.c , I can help in making the pinmux to pinctrl transition.

**Comment by mbrunnen** [comment]
> > @ithinx-io - any thoughts on adding pinctrl support? @vaishnavachath - are you able to offer any advice there?
> 
> @ithinx-io You can take reference from https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pinctrl/pinctrl_cc13xx_cc26xx.c , I can help in making the pinmux to pinctrl transition.

Thanks a lot, that's a very useful pointer.

**Pull request closed without merging**